src/components/Buttons/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,43 @@
+import { SetStateAction } from "react";
+import style from "./Checkbox.module.scss";
+import { RegisterCheckboxValue } from "@/services/types";
+
+interface ICheckbox {
+  label: string;
+  initialValue: RegisterCheckboxValue;
+  registerName: string;
+  setValue: (name: string, value: SetStateAction<RegisterCheckboxValue>) => void;
+  getValues: (name: string) => any;
```

Comment:
**nit:** You should not need that here, you can check it from parent whenever checkbox is checked or not